### ISSUE TITLE
Issue with calendar loading

### ISSUE THREAD

**Issue by MedaiP90** [twinssbc/Ionic2-Calendar#532]
Since the last update (ionic2-calendar version 0.6.2), when I access the calendar page in my android app everything is ok, but when I navigate to another page, change the source data and return to the calendar page the calendar itself (in monthview mode) load only one month with no events in it and this error appear:
```
main-es2015.5f676f4d708acda70489.js:1 ERROR TypeError: Cannot read property 'dates' of undefined
    at Z.onDataLoaded (100-es2015.242a26d3525d1abf3066.js:1)
    at Z.ngOnChanges (100-es2015.242a26d3525d1abf3066.js:1)
    at Z.e.type.ngOnChanges.e.onChanges (main-es2015.5f676f4d708acda70489.js:1)
    at fn (main-es2015.5f676f4d708acda70489.js:1)
    at hn (main-es2015.5f676f4d708acda70489.js:1)
    at cn (main-es2015.5f676f4d708acda70489.js:1)
    at Gi (main-es2015.5f676f4d708acda70489.js:1)
    at main-es2015.5f676f4d708acda70489.js:1
    at Gi (main-es2015.5f676f4d708acda70489.js:1)
    at Eo (main-es2015.5f676f4d708acda70489.js:1)
```

**Comment by twinssbc** [comment]
@MedaiP90 Could you provide more code? How do you navigate between the pages and where do you change the source data?

**Comment by MedaiP90** [comment]
> @MedaiP90 Could you provide more code? How do you navigate between the pages and where do you change the source data?

The events data are updated retriving them from firebase but I think the problem is in the monthview template override:
```html
[...]
    <calendar
        (onTimeSelected)="dateSelected($event)"
        (onTitleChanged)="onViewTitleChanged($event)"
        [currentDate]="currentDate"
        [eventSource]="eventSource"
        [locale]="locale"
        [monthviewDisplayEventTemplate]="calendarActiveEvents"
        [monthviewEventDetailTemplate]="template"
        [monthviewInactiveDisplayEventTemplate]="calendarActiveEvents"
        calendarMode="month"
        endHour="20"
        startHour="6"
        startingDayMonth="1"
        step="30"
    >
    </calendar>

[...]

<ng-template #calendarActiveEvents let-view="view" let-row="row" let-col="col">
    <div *ngIf="!view.dates[row*7+col].secondary" class="cell generic-day">
        <!-- Month day -->
        <div
            [class.current-day]="view.dates[row*7+col].current"
            [class.selected-day]="view.dates[row*7+col].selected"
            [class.non-selected-day]="!view.dates[row*7+col].selected && !view.dates[row*7+col].current"
        >
            <div class="day-title">{{ view.dates[row*7+col].label }}</div>
            <div class="circles-container">
                <div
                    class="circle"
                    *ngFor="let color of getCellClass(view.dates[row*7+col])"
                    [style.background-color]="color">
                </div>
                <div
                    *ngIf="getCellClass(view.dates[row*7+col]).length === 0"
                    class="circle"
                    [style.background-color]="'white'">
                </div>
            </div>
        </div>
    </div>
    <div *ngIf="view.dates[row*7+col].secondary" class="cell secondary-day">
        <!-- Other month day -->
        <div class="day-title">{{ view.dates[row*7+col].label }}</div>
        <div class="circles-container">
            <div
                class="circle"
                *ngFor="let color of getCellClass(view.dates[row*7+col])"
                [style.background-color]="'white'">
            </div>
        </div>
    </div>
</ng-template>
```

**Comment by MedaiP90** [comment]
And here they are some pictures of the problem: the first is when the calendar correctly load the first time I open the page, the second one is when I change the source data and navigate back to the calendar and the third is when I swipe to change month. I wound like to point out that this issue has never shown up before the "ivy" update of the calendar package.

![No problem](https://user-images.githubusercontent.com/20813764/80342124-e5659100-8863-11ea-854f-d80bfb90b573.jpg)
![Source data changed](https://user-images.githubusercontent.com/20813764/80342133-e8608180-8863-11ea-82d5-2ead81a5aeb8.jpg)
![Swiped to change month](https://user-images.githubusercontent.com/20813764/80342139-eac2db80-8863-11ea-909c-b71b81d60041.jpg)

**Comment by massimodeluisa** [comment]
Quote @MedaiP90, same issue.

**Comment by MGaden** [comment]
i had this issue.
i solved it by change in file (monthview.js) as below image
![Annotation 2020-05-01 162013](https://user-images.githubusercontent.com/15795301/80812259-de25e680-8bc7-11ea-85c8-dad510608d35.png)

the issue is (this.currentViewIndex) is null, i don't know why. but this solve my issue

**Comment by twinssbc** [comment]
@MedaiP90 @MGaden I'm still not able to reproduce this issue. But based on @MGaden 's code, I also add some fix accordingly. Could you give a try? The latest version is 0.6.3.

Also if you could continue sharing more information, that will be helpful. May I know when the event retrieving happen in your code? Could you provide your component ts code?

**Comment by MedaiP90** [comment]
@twinssbc I'll give a try to the new **0.6.3** version. Anyway this is my component code:
```typescript
@Component({
    selector: 'app-calendar',
    templateUrl: './calendar.page.html',
    styleUrls: ['./calendar.page.scss'],
})
export class CalendarPage extends AbstractAnalyticsPage {

    // Public properties

    public eventSource: IEvent[];
    public currentDate: Date;
    public calendarTitle: string;
    public locale: string;
    public selectedMonth: string;
    public selectedCollectionDate: Date;

    // Private properties

    private zoneStreamSubscription: Subscription;
    private readonly currentCurrentZoneService: CurrentZoneService;
    private readonly defaultEventEndHours: number;
    private readonly defaultEventStartHours: number;
    private readonly translateService: TranslateService;

    public constructor(
        currentCurrentZoneService: CurrentZoneService,
        translateService: TranslateService,
    ) {
        super({ pageName: 'calendar' });
        this.currentCurrentZoneService = currentCurrentZoneService;
        this.translateService = translateService;
        this.defaultEventStartHours = 9;
        this.defaultEventEndHours = 21;
        this.eventSource = [];
        this.setupLocale();
    }

    // Life cycle hooks

    public async ionViewWillEnter(): Promise<void> {
        this.setupLocale();
        this.currentDate = new Date();

        super.ionViewWillEnter();
    }

    public async ionViewDidEnter(): Promise<void> {
        await super.ionViewDidEnter();
        this.zoneStreamSubscription = this.currentCurrentZoneService
            .currentZone
            .subscribe(async (zone: IZone): Promise<void> => {
                this.eventSource = [];

                if (zone === undefined || zone.collections === undefined || zone.collections.length === 0) {
                    return;
                }

                for (const collection of zone.collections) {
                    const event: IEvent = collection as IEvent;

                    event.allDay = true;
                    event.startTime = collection.when.toDate();
                    event.startTime.setHours(this.defaultEventStartHours);
                    event.endTime = collection.when.toDate();
                    event.endTime.setHours(this.defaultEventEndHours);

                    this.eventSource.push(event);
                }
            });
    }

    public ionViewDidLeave(): void {
        super.ionViewDidLeave();
        if (this.zoneStreamSubscription instanceof Subscription) { this.zoneStreamSubscription.unsubscribe(); }
    }

    // Public methods

    public onViewTitleChanged(title: string): void {
        this.calendarTitle = title;
    }

    public dateSelected(timeSelected: ITimeSelected): void {
        this.selectedCollectionDate = timeSelected.selectedTime;
    }

    public getCellClass(monthviewCell: IMonthviewCell): string[] {
       [...]
    }

    // Private methods

    private setupLocale(): void {
        this.locale = this.translateService.currentLang;
    }
}

```

**Comment by MedaiP90** [comment]
@twinssbc the update hasn't solved the problem for me and not even @MGaden solution did. I'm using capacitor instead of cordova (if that could help) and a tab based view to navigate between pages. I've inserted a `console.log(this.currentViewIndex)` in the `onDataLoaded()` method and I found this: the first load do what it's supposed to, it loads the data and update it when swiping; the second time I enter the calendar page, `onDataLoaded()` is called two times at start (loading nothing) and never called again while swiping.

**Comment by twinssbc** [comment]
@MedaiP90 Today I was able to reproduce this error and then started looking into the ion slides source code.
Then I encounter this commit. 
https://github.com/ionic-team/ionic/commit/3123a318b6755dbf8fde8476f81c37e6ffa9b70e#diff-e0874821af764a71e89442a158a223f8

After I upgrade the @ionic/angular to 5.1.0, seems the issue is solved.
Could you give a try? If that also works for you, I will upgrade the ionic dependency to 5.1.0.

**Comment by MedaiP90** [comment]
@twinssbc thank you, updating `@ionic/angular` to **5.1.0** solved the problem

**Comment by massimodeluisa** [comment]
@twinssbc same for me!

**Comment by SilverMountainSoftware** [comment]
I'm using @ionic/angular 5.0.7 and your 0.6.3 fixed it for me. I will update all of my packages today

**Comment by twinssbc** [comment]
I have published a version 0.6.4 which requires using Ionic 5.1.0 onwards, feel free to give a try.

**Comment by SilverMountainSoftware** [comment]
Yes, this fix worked well for me, thanks
